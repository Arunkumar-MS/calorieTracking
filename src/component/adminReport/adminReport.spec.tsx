import { jest } from '@jest/globals'
import { shallowToJson } from "enzyme-to-json";
import * as React from 'react';
import { shallow } from 'enzyme';
import AdminReport from './index';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: () => mockDispatch
}));


jest.mock("swr", () => ({
    __esModule: true,
    namedExport: jest.fn(),
    default: () => ({
        data: { "avgCalForLastSevendays": [{ "date": "02:02:2022", "value": 839.6300000000001 }, { "date": "01:02:2022", "value": 1508.3500000000001 }, { "date": "03:02:2022", "value": 11737.93 }, { "date": "29:01:2022", "value": 303.4 }, { "date": "30:01:2022", "value": 189 }, { "date": "31:01:2022", "value": 210.5 }], "lastSevenDaysEntries": 83, "weekBeforeLastSevenDaysEntries": 7 },
    }),
}));

describe('AdminReport page', () => {
    it('should render properly', function () {
        const wrap = shallow(<AdminReport />)
        expect(shallowToJson(wrap)).toMatchSnapshot();
    })
})
