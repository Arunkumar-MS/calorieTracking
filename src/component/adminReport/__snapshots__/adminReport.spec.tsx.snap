// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminReport page should render properly 1`] = `
<Fragment>
  <div
    className="mt-5"
  >
    <ForwardRef(LoadableComponent)
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "gray",
              "data": Array [
                210.5,
                189,
                303.4,
                11737.93,
                1508.3500000000001,
                839.6300000000001,
              ],
            },
          ],
          "labels": Array [
            "31:01:2022",
            "30:01:2022",
            "29:01:2022",
            "03:02:2022",
            "01:02:2022",
            "02:02:2022",
          ],
        }
      }
      options={
        Object {
          "plugins": Object {
            "legend": null,
            "title": Object {
              "display": true,
              "text": "Last 7 days avarage per user calories consumption per day",
            },
          },
          "responsive": true,
        }
      }
    />
    <div
      className="mt-20"
    >
      <ForwardRef(LoadableComponent)
        data={
          Object {
            "datasets": Array [
              Object {
                "backgroundColor": Array [
                  "rgba(255, 99, 132, 0.2)",
                  "rgba(54, 162, 235, 0.2)",
                ],
                "borderColor": Array [
                  "rgba(255, 99, 132, 1)",
                  "rgba(54, 162, 235, 1)",
                ],
                "borderWidth": 1,
                "data": Array [
                  7,
                  83,
                ],
              },
            ],
            "labels": Array [
              "Week before 7 days entry",
              "Last 7 days entry",
            ],
          }
        }
        options={
          Object {
            "plugins": Object {
              "legend": null,
              "title": Object {
                "display": true,
                "text": "Last 7 days user entry vs week before 7 days entry",
              },
            },
            "responsive": true,
          }
        }
      />
    </div>
  </div>
</Fragment>
`;
